cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -364,40 +374,65 @@ public void apply(MetadataDataset input) throws Exception {
   }
 
   @Override
-  public Set<MetadataSearchResultRecord> searchMetadata(String namespaceId, String searchQuery) {
-    return ImmutableSet.<MetadataSearchResultRecord>builder()
-      .addAll(searchMetadata(MetadataScope.USER, namespaceId, searchQuery))
-      .addAll(searchMetadata(MetadataScope.SYSTEM, namespaceId, searchQuery))
-      .build();
-  }
-
-  @Override
-  public Set<MetadataSearchResultRecord> searchMetadata(MetadataScope scope, String namespaceId, String searchQuery) {
-    return searchMetadataOnType(scope, namespaceId, searchQuery, ImmutableSet.of(MetadataSearchTargetType.ALL));
-  }
+  public Set<MetadataSearchResultRecord> search(String namespaceId, String searchQuery,
+                                                Set<MetadataSearchTargetType> types,
+                                                @Nullable String sort) throws BadRequestException {
+    if ("*".equals(searchQuery)) {
+      if (!Strings.isNullOrEmpty(sort)) {
+        return search(MetadataScope.SYSTEM, namespaceId, searchQuery, types, getSortInfo(sort));
+      }
+      // Can't disallow this completely, because it is required for upgrade, but log a warning to indicate that
+      // a full index search should not be done in production.
+      LOG.warn("Attempt to search through all indexes. This query can have an adverse effect on performance and is " +
```

Comment:
did we purposely take the empty string check out of the http handler? The handler isn't called by the upgrade tool is it?